model-zoo/src/main/java/ai/djl/basicmodelzoo/cv/object_detection/yolo/YOLOV3.java

```diff
@@ -0,0 +1,417 @@
+
+
+    public static Block makeLastLayers(int filtersIn,int filtersOut,float batchNormMomentum,float leakyAlpha){    //filters[0], filters[1]
+        return new SequentialBlock().add(
+                convolutionBlock(filtersIn,1,batchNormMomentum,leakyAlpha)
+        ).add(
+                convolutionBlock(filtersOut,3,batchNormMomentum,leakyAlpha)
+        ).add(
+                convolutionBlock(filtersIn,1,batchNormMomentum,leakyAlpha)
+        ).add(
+                convolutionBlock(filtersOut,3,batchNormMomentum,leakyAlpha)
+        ).add(
+                convolutionBlock(filtersIn,1,batchNormMomentum,leakyAlpha)
+        );
+    }
+
+    public static Block makeOutputLayers(int filtersOut,int outClass,float batchNormMomentum,float leakyAlpha){
+        return new SequentialBlock().add(
+                convolutionBlock(filtersOut,3,batchNormMomentum,leakyAlpha)
+        ).add(
+                Conv2d.builder().setFilters(outClass)
+                        .setKernelShape(new Shape(1,1))
+                        .build()
+        );
+    }
+
+    @Override
+    protected NDList forwardInternal(ParameterStore parameterStore, NDList inputs, boolean training, PairList<String, Object> params) {
+        System.out.println("In forwardInternal");
+        NDList xList = darkNet53.forward(parameterStore,inputs,training);
+        System.out.println(xList);
+        NDArray x0 = xList.get(0),x1 = xList.get(1), x2 = xList.get(2);
+
+        //the first feature layer
+        System.out.println(x0.getShape());
+        System.out.println(x1.getShape());
+        System.out.println(x2.getShape());
+
+
+        NDList out0Branch = lastLayer0.forward(parameterStore,new NDList(x0),training);
+        NDList out0 = layer0Output.forward(parameterStore,out0Branch,training);
+
+        NDList x1In = lastLayer1Conv.forward(parameterStore,out0Branch,training);
+        x1In = lastLayer1UpSample.forward(parameterStore,x1In,training);
+        System.out.println(x1In);
+        x1In = new NDList(x1In.singletonOrThrow().concat(x1,1));
+
+        //the second feature Layer
+        NDList out1Branch = lastLayer1.forward(parameterStore,x1In,training);
+        NDList out1 = layer1Output.forward(parameterStore,out1Branch,training);
+
+        NDList x2In = lastLayer2Conv.forward(parameterStore,out1Branch,training);
+        x2In = lastLayer2UpSample.forward(parameterStore,x2In,training);
+        x2In = new NDList(x2In.singletonOrThrow().concat(x2,1));
+        System.out.println(Arrays.toString(x2In.getShapes()));
+
+
+        //the third feature layer
+        NDList out2 = lastLayer2.forward(parameterStore,x2In,training);
+        out2 = layer2Output.forward(parameterStore,out2,training);
+
+        //Outputs
+        return new NDList(
+                out0.singletonOrThrow(),
+                out1.singletonOrThrow(),
+                out2.singletonOrThrow()
+        );
+    }
+
+    @Override
+    public Shape[] getOutputShapes(Shape[] inputShapes) {
```

Comment:
Is this right? I would imagine that the logic here would be similar to forward if you are doing the computation fully, or could skip steps to replace computation with fixed knowledge. This looks a lot like a SequentialBlock logic, but I don't remember Yolo being sequential. Similar applies to the initializeChildBlocks which should also be similar to forward as it has to pass the right type of inputs to each intermediate block